src/query/trigger_context.hpp

```diff
@@ -65,6 +65,11 @@ struct DeletedObject {
   TAccessor object;
 };
 
+template <>
+inline bool DeletedObject<EdgeAccessor>::IsValid() const {
+  return object.StorePropertiesOnEdges() && object.IsVisible(storage::View::OLD);
+}
+
```

Comment:
This solves the crash but introduces a more subtle bug with isolation levels. The issue that I think it introduces is with visibility of objects in triggers. As you can see in `IsVisible` we are iterating through deltas when edges have properties since they have deltas as well then. However, even when edges don't have properties they still have deltas, their isolation is still dictated by mvcc. The issue is that their deltas are within start and end nodes, basically, you would have to iterate through start or end nodes (It does not matter which one, since deltas are stored in both) and look for the following deltas:
- `ADD_IN_EDGE`
- `ADD_OUT_EDGE`
- `REMOVE_IN_EDGE`
- `REMOVE_OUT_EDGE`
which can be applied on edges